### PR TITLE
Datasets, state files and scripts for DMS paper

### PR DESCRIPTION
**Pull request by julesvidal**
Dear Julien, 
This PR contains the state files, data, scripts and documentation for the experiments from the DMS paper.
Additionally, I modified the `run.py` script in `tests/`, to be able to specify a resolution factor from the commandline, in order to produce high-res pictures from pvsm files.
I'll prepare some screenshots of the paraview states for the website. 
Cheers,
Jules

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -254,3 +254,52 @@ Data specs
   "acknowledgement": "Berk Geveci, Christoph Garth, Garrett Aldrich, and Jonas Lukasczyk",
   "url": "https://jluk.de/resources/papers/NestedTrackingGraphs2017.pdf"
 }
+
+{
+  "name": "at.vti",
```
Comment (julien-tierny) [inline README]: could you please insert these items at the right spot in the file?
(the entries are supposed to be in alphabetical order of their name).

File: README [hunk 2]
```diff
@@ -254,3 +254,52 @@ Data specs
...
+  "description": "Quantum chemistry dataset. Simulation of the electronic density in the Adenine-Thymine molecular complex",
+  "acknowledgement": "",
```
Comment (julien-tierny) [inline README]: "acknowledgement": "Roberto Alvarez Boto",

File: README [hunk 3]
```diff
@@ -254,3 +254,52 @@ Data specs
...
+  "url": ""
+}
+
+{
+  "name": "ds14_scivis_0128_e4_dt04_1.0000.vtp",
+  "description": "Point cloud dataset, simulation of the density of dark matter in the universe",
+  "acknowledgement": "",
+  "url": ""
```
Comment (julien-tierny) [inline README]: please check mathieu's paper to get the URL to the right scivis contest: https://arxiv.org/pdf/2107.07789

File: README [hunk 4]
```diff
@@ -254,3 +254,52 @@ Data specs
...
+  "name": "at.vti",
+  "description": "Quantum chemistry dataset. Simulation of the electronic density in the Adenine-Thymine molecular complex",
+  "acknowledgement": "",
+  "url": ""
+}
+
+{
+  "name": "ds14_scivis_0128_e4_dt04_1.0000.vtp",
+  "description": "Point cloud dataset, simulation of the density of dark matter in the universe",
+  "acknowledgement": "",
```
Comment (julien-tierny) [inline README]: "IEEE SciVis Contest 20xx"
xx to be determined by looking at mathieu's paper: https://arxiv.org/pdf/2107.07789

File: README [hunk 5]
```diff
@@ -254,3 +254,52 @@ Data specs
...
+  "url": ""
+}
+
+{
+  "name": "ferility.vtu",
+  "description": "2d triangulation of the *fertility* dataset. Scanning and reconstruction.",
+  "acknowledgement": "",
```
Comment (julien-tierny) [inline README]: "acknowledgement": "AIM@Shape",

File: README [hunk 6]
```diff
@@ -254,3 +254,52 @@ Data specs
...
+  "url": "http://visionair.ge.imati.cnr.it/ontologies/shapes/view.jsp?id=670-fertility_-_watertight"
+}
+
+{
+  "name": "casting.vtu",
+  "description": "2d triangulation of the surface of a mechanical piece",
+  "acknowledgement": "",
```
Comment (julien-tierny) [inline README]: "acknowledgement": "AIM@Shape",

File: README [hunk 7]
```diff
@@ -254,3 +254,52 @@ Data specs
...
+  "description": "3d triangulation of the shape of a skull",
+  "acknowledgement": "",
+  "url": ""
+}
+
+{
+  "name": "periodicPicture.cdb",
+  "description": "Cinema database containing 100 greyscale pictures with resolution 32x32, of a synthetic terrain, with 100 viewpoints arranged along a circle. Generated using ParaView",
+  "acknowledgement": "",
+  "url": ""
```
Comment (julien-tierny) [inline README]: "url": "https://github.com/topology-tool-kit/ttk-data"

File: README [hunk 8]
```diff
@@ -254,3 +254,52 @@ Data specs
...
+  "name": "skull.vtu",
+  "description": "3d triangulation of the shape of a skull",
+  "acknowledgement": "",
+  "url": ""
+}
+
+{
+  "name": "periodicPicture.cdb",
+  "description": "Cinema database containing 100 greyscale pictures with resolution 32x32, of a synthetic terrain, with 100 viewpoints arranged along a circle. Generated using ParaView",
+  "acknowledgement": "",
```
Comment (julien-tierny) [inline README]: "acknowledgement": "Jules Vidal",

File: docs/persistentGenerators_at.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+## Inputs
+- [at.vti](https://github.com/topology-tool-kit/ttk-data/raw/dev/at.vti): A molecular dataset: a three-dimensional regular grid with 1 scalar field, the electronic density in the Adenine Thymine complex.
+
+## Outputs
+-  `PersistentGeneratorsAt.vtp`: the raw one-dimensional generators of the field.
+
+
+## C++/Python API
+
+[PersistentGenerators](https://topology-tool-kit.github.io/doc/html/classttkPersistentGenerators.html)
```
Comment (julien-tierny) [inline docs/persistentGenerators_at.md]: can you please make sure that **all** the TTK modules used in the corresponding python script are indeed listed here? 
(I have the impression that a few mkdocs files for this PR are missing a few TTK modules).
also, can you make sure that the entries are listed in alphabetical order?

File: docs/persistentGenerators_householdAnalysis.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
+# Persistent Generators Household Analysis
+
+![Persistent Generators Household Analysis example Image](https://topology-tool-kit.github.io/img/gallery/persistentGenerators_householdAnalysis.jpg)
+
+## Pipeline description
+
+This example shows the detection of a circular pattern in hiigh-dimensional data.
```
Comment (julien-tierny) [inline docs/persistentGenerators_householdAnalysis.md]: hiigh --> high

File: docs/persistentGenerators_periodicPicture.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
+# Persistent Generators Periodic Picture
+
+![Persistent Generators Periodic Picture example Image](https://topology-tool-kit.github.io/img/gallery/persistentGenerators_periodicPicture.jpg)
+
+## Pipeline description
+
+This example shows the detection of a circular pattern in hiigh-dimensional data.
```
Comment (julien-tierny) [inline docs/persistentGenerators_periodicPicture.md]: hiigh -> high

File: docs/persistentGenerators_periodicPicture.md [hunk 12]
```diff
@@ -0,0 +1,53 @@
...
+- [periodicPicture.cdb](https://github.com/topology-tool-kit/ttk-data/raw/dev/periodicPicture.cdb): A cinema database containing all 100 pictures, with metadata (the associated camera angle).
+
+## Outputs
+-  `PersistentGeneratorsPeriodicPicture_points.vtp`: the high-dimensionnal point cloud embedded in 3D.
+-  `PersistentGeneratorsPeriodicPicture_ripsComplex.vtu`: the 2-dimensional Rips complex of the point cloud.
+-  `PersistentGeneratorsPeriodicPicture_cycle.vtp`: the infinitely persistent 1-cycle of the Rips complex.
+
+
+## C++/Python API
+
```
Comment (julien-tierny) [inline docs/persistentGenerators_periodicPicture.md:45]: back to my previous comment, the TTK cinema filters are not referenced here.
(thanks for listing all the TTK modules used in the python script).

also, thanks for putting them in alphabetical order.

File: python/persistentGenerators_householdAnalysis.py [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+
+# create a new 'K Means'
+kMeans1 = KMeans(Input=tableToPoints1,
+    ModelInput=None)
+kMeans1.VariablesofInterest = ['Component_0', 'Component_1', 'Component_2']
+kMeans1.k = 8
+
+# create a new 'TTK RipsComplex'
+tTKRipsComplex1 = TTKRipsComplex(Input=tTKTableDistanceMatrix1)
+tTKRipsComplex1.SelectFieldswithaRegexp = 1
```
Comment (julien-tierny) [inline python/persistentGenerators_householdAnalysis.py:35]: is the (long) specification of the input columns (below) required?
isn't the regexp sufficient?

in other words, does the script work if you remove the next line?

File: python/persistentGenerators_periodicPicture.py [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
... '0874', '0875', '0876', '0877', '0878', '0879', '0880', '0881', '0882', '0883', '0884', '0885', '0886', '0887', '0888', '0889', '0890', '0891', '0892', '0893', '0894', '0895', '0896', '0897', '0898', '0899', '0900', '0901', '0902', '0903', '0904', '0905', '0906', '0907', '0908', '0909', '0910', '0911', '0912', '0913', '0914', '0915', '0916', '0917', '0918', '0919', '0920', '0921', '0922', '0923', '0924', '0925', '0926', '0927', '0928', '0929', '0930', '0931', '0932', '0933', '0934', '0935', '0936', '0937', '0938', '0939', '0940', '0941', '0942', '0943', '0944', '0945', '0946', '0947', '0948', '0949', '0950', '0951', '0952', '0953', '0954', '0955', '0956', '0957', '0958', '0959', '0960', '0961', '0962', '0963', '0964', '0965', '0966', '0967', '0968', '0969', '0970', '0971', '0972', '0973', '0974', '0975', '0976', '0977', '0978', '0979', '0980', '0981', '0982', '0983', '0984', '0985', '0986', '0987', '0988', '0989', '0990', '0991', '0992', '0993', '0994', '0995', '0996', '0997', '0998', '0999', '1000', '1001', '1002', '1003', '1004', '1005', '1006', '1007', '1008', '1009', '1010', '1011', '1012', '1013', '1014', '1015', '1016', '1017', '1018', '1019', '1020', '1021', '1022', '1023']
```
Comment (julien-tierny) [inline python/persistentGenerators_periodicPicture.py]: cf my above remark:

"
is the (long) specification of the input columns (below) required?
isn't the regexp sufficient?

in other words, does the script work if you remove the next line?
"

**Review by julien-tierny — CHANGES_REQUESTED** [review]
thanks a lot jules!
that's great stuff!

could you please address the few remarks I entered in my review?
thanks!

**Comment by julesvidal** [comment]
Hi Julien, 
I addressed your comments. 
Just a remark : the documentation links in multiple places to https://topology-tool-kit.github.io/doc/html/classttkPersistentGenerators.html, which is not created yet. Should I remove these links ?
cheers, 
jules

**Comment by julien-tierny** [comment]
thanks a lot jules!
(and thanks for the heads up on the dead link: I'd suggest to keep things this way for the moment, I'll update the doxygen online documentation asap)

**Comment by julien-tierny** [comment]
awesome, let's go!

**Comment by julien-tierny** [comment]
One more thing: 
could you please go over the source code of **each module** used in each of these examples to insert (in the paraview xml file, the vtk header, the base class header, all in alphabetical order) the link to the corresponding example pages?

here are an example for the topological simplification:
- https://github.com/topology-tool-kit/ttk/blob/dev/paraview/xmls/TopologicalSimplification.xml
- https://github.com/topology-tool-kit/ttk/blob/dev/core/vtk/ttkTopologicalSimplification/ttkTopologicalSimplification.h
- https://github.com/topology-tool-kit/ttk/blob/dev/core/base/topologicalSimplification/TopologicalSimplification.h

thanks!

**Pull request merged**